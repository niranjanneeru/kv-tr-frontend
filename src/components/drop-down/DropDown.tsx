import './style.css';

const DropDown = ({ label, options, placeHolder, value, onChangeCallback }) => {
	console.log(value);

	return (
		<div className='input-group'>
			<label className='dropdown-label'>{label}</label>
			<select className='normal-input' id='status' onChange={onChangeCallback}>
				{value === '' && (
					<option value='' selected disabled hidden>
						{placeHolder}
					</option>
				)}
				{Object.keys(options).map((key) => {
					return (
						<option key={key} value={key} selected={value !== '' && key === value ? true : false}>
							{options[key]}
						</option>
					);
				})}
			</select>
		</div>
	);
};

export default DropDown;
