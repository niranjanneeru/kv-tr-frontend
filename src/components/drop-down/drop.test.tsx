import { render, screen, fireEvent } from "@testing-library/react";
import DropDown from "./DropDown";
import '@testing-library/jest-dom'

describe("Test for DropDown Component", () => {
    const options = {
        option1: "Option 1",
        option2: "Option 2",
        option3: "Option 3",
    };

    it("Renders correctly with label and placeholder", () => {
        render(
            <DropDown
                label="Select an option"
                options={options}
                placeHolder="Choose an option"
                value=""
                onChangeCallback={() => { }}
            />
        );

        const labelElement = screen.getByText("Select an option");
        const placeholderOption = screen.getByText("Choose an option");

        expect(labelElement).toBeInTheDocument();
        expect(placeholderOption).toBeInTheDocument();
    });

    it("Renders options correctly", () => {
        render(
            <DropDown
                label="Select an option"
                options={options}
                placeHolder="Choose an option"
                value=""
                onChangeCallback={() => { }}
            />
        );

        const option1 = screen.getByText("Option 1");
        const option2 = screen.getByText("Option 2");
        const option3 = screen.getByText("Option 3");

        expect(option1).toBeInTheDocument();
        expect(option2).toBeInTheDocument();
        expect(option3).toBeInTheDocument();
    });
});
