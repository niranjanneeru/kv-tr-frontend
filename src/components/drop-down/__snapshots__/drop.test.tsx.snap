// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for DropDown Component To Match Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="input-group"
      >
        <label
          class="dropdown-label"
        >
          Select an option
        </label>
        <select
          class="normal-input"
          id="status"
        >
          <option
            disabled=""
            hidden=""
            value=""
          >
            Choose an option
          </option>
          <option
            value="option1"
          >
            Option 1
          </option>
          <option
            value="option2"
          >
            Option 2
          </option>
          <option
            value="option3"
          >
            Option 3
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="input-group"
    >
      <label
        class="dropdown-label"
      >
        Select an option
      </label>
      <select
        class="normal-input"
        id="status"
      >
        <option
          disabled=""
          hidden=""
          value=""
        >
          Choose an option
        </option>
        <option
          value="option1"
        >
          Option 1
        </option>
        <option
          value="option2"
        >
          Option 2
        </option>
        <option
          value="option3"
        >
          Option 3
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
